### PR TITLE
Implement changes for start and end date time for Free Command

### LINKED ISSUE #120: [PE-D] Past time interval can be taken as free
Reproduce step:
free sd/2021-01-01 10:00 ed/2021-01-01 12:00
Expected:
The time interval provided is past
Actual:
You're free!

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: ljhgab/ped#6

### LINKED ISSUE #121: [PE-D] Free command end date can be before start date
Step to reproduce:
free sd/ 2021-04-03 12:00  ed/2021-01-01 10:00
Expected: End date should be after start date
Actual: No error message produced

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: ljhgab/ped#7

### LINKED ISSUE #124: [PE-D] Entry with only endDate is considered in checking free time
Step to reproduce:
With one task that ends on 2021-02-15 23:00
free sd/ 2021-02-15 22:00 ed/ 2021-02-16 08:00

Expected:
Sorry you are not free.
Actual:
You are free!

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: ljhgab/ped#10

### LINKED ISSUE #128: [PE-D] 'free' command doesn't restrict user from keying in an end-date that is before start-date
User can key in the shown command and it will be executed. Output is "You're free!"

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Zaiah0505/ped#2

### LINKED ISSUE #143: [PE-D] Bug: time error message
Error message is wrong, can change it to the starting time is after the end time

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: VisnuRavi/ped#5

### LINKED ISSUE #146: [PE-D] Bug: free start date after end date
Entering a start date after the end date results in a success message, can look to rectifying that

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: VisnuRavi/ped#8

### LINKED ISSUE #147: [PE-D] Bug: start time after end time
Entering a start time after the end time in free is acceptable

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: VisnuRavi/ped#9

### LINKED ISSUE #165: Tests for free command for entries

### PR DESCRIPTION
**Pull request by tsh22**
* Ensure start time cannot be after end time
* Ensure past time interval cannot be provided for free command
* Change error msg phrasing for invalid date range: Specify that either date or time is the cause of the error instead of just date
* Change usage message for free command
* Testing for FreeCommand, FreeCommandParser and EntryDate

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 1]
```diff
@@ -9,7 +9,9 @@
     public static final String MESSAGE_INVALID_ENTRY_INDEX = "The entry index provided is invalid";
     public static final String MESSAGE_DELETE_ENTRY_SUCCESS = "Deleted Entry: %1$s";
     public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
-    public static final String MESSAGE_INVALID_DATE_RANGE = "The provided start date is after the provided end date!";
+    public static final String MESSAGE_INVALID_DATE_RANGE = "The provided start date or time is after the provided"
+            + " end date or time!";
+    public static final String MESSAGE_DATE_IN_PAST = "Time travel is not allowed! Please provide a valid date time!";
```
Comment (nicholastanvis) [inline src/main/java/seedu/address/commons/core/Messages.java]: let's change this to MESSAGE_ENTRY_START_DATE_IN_PAST

**Review by nicholastanvis — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/commons/core/Messages.java (hunk 1 above)
Comment (nicholastanvis) [reply src/main/java/seedu/address/commons/core/Messages.java]: also use "end date and time", "or time" is kinda weird

Comment (tsh22) [reply src/main/java/seedu/address/commons/core/Messages.java]: Is it ok to stick with or time? Cause i wanna try to indicate that it's either date or time that is not a valid range. Could be that start time is after end time or start date is after end date, or is there a better phrasing?

Comment (nicholastanvis) [reply src/main/java/seedu/address/commons/core/Messages.java]: heh, i look at it like "date and time" is a phrase. anything ig

Comment (tsh22) [reply src/main/java/seedu/address/commons/core/Messages.java]: Ok i changed it a little, see if it's less weird like that!

File: src/main/java/seedu/address/logic/commands/FreeCommand.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Indicates if an interval is free. "
             + "If free, 'free' will be shown. If not, tasks occupying that interval will be shown in "
-            + "this result display\n"
+            + "the entries list below.\n"
```
Comment (nicholastanvis) [inline src/main/java/seedu/address/logic/commands/FreeCommand.java]: - entry list
- entries listed
- list

choose your poison

**Review by nicholastanvis — APPROVED** [review]

**Review by lrj689 — APPROVED** [review]

**Review by nicholastanvis — APPROVED** [review]

**Review by goatygoatygoat — APPROVED** [review]

**Pull request merged**